packages/react-widgets/src/widgets/CategoryWidget.js

```diff
@@ -52,6 +52,8 @@ function CategoryWidget(props) {
   } = props;
   const dispatch = useDispatch();
 
+  const source = useSelector((state) => selectSourceById(state, dataSource))
+  const isDroppingFeatures = source?.isDroppingFeatures
```

Comment:
I would create a specific selector for dropping features property so we don't have to declare a variable with the full source that may change a lot. Maybe, you can put that logic inside useWidgetFetch where we are already have the source object. Take a look into that hook.